demo/midgard-sdk/src/settlement.ts

```diff
@@ -0,0 +1,513 @@
+        cause: "No UTxOs found in Hub Oracle contract address",
+      });
+    }
+    const hubOracleRefUTxOs = yield* utxosToHubOracleUTxOs(allUTxOs, hubOraclePolicyId);
+    return hubOracleRefUTxOs[0];
+  });
+
+const fetchSchedulerRefUTxOs = (
+  schedulerScriptAddress: string,
+  schedulerPolicyId: string,
+  lucid: LucidEvolution,
+): Effect.Effect<SchedulerUTxO, LucidError> =>
+  Effect.gen(function* () {
+    const allUTxOs = yield* Effect.tryPromise({
+      try: () => lucid.utxosAt(schedulerScriptAddress),
+      catch: (err) =>
+        new LucidError({
+          message: "Failed to fetch Scheduler UTxOs",
+          cause: err,
+        }),
+    });
+    if (allUTxOs.length === 0) {
+      yield* new LucidError({
+        message: "Failed to build the Scheduler transaction",
+        cause: "No UTxOs found in Scheduler contract address",
+      });
+    }
+    const schedulerRefUTxO = yield* utxosToSchedulerUTxOs(allUTxOs, schedulerPolicyId);
+    return schedulerRefUTxO[0];
+  });
+
+/**
+ * Settlement
+ *
+ * @param lucid - The LucidEvolution
+ * @param params - The parameters
+ * @returns {TxBuilder} A TxBuilder instance that can be used to build the transaction.
+ */
+export const incompleteAttachResolutionClaimTxProgram = (
+  lucid: LucidEvolution,
+  params: AttachResolutionClaimParams,
+): Effect.Effect<TxBuilder, HashingError | DataCoercionError | LucidError> =>
+  Effect.gen(function* () {
+    const spendRedeemer: SettlementSpendRedeemer = {
+      AttachResolutionClaim: {
+        settlementInputIndex: 0n,
+        settlementOutputIndex: 0n,
+        hubRefInputIndex: 0n,
+        activeOperatorsNodeInputIndex: 0n,
+        activeOperatorsRedeemerIndex: 0n,
+        operator: params.resolutionClaimOperator,
+        schedulerRefInputIndex: 0n,
+      },
+    };
+    const spendRedeemerCBOR = Data.to(spendRedeemer, SettlementSpendRedeemer);
+
+    const settlementAllUtxos: UTxO[] = yield* Effect.tryPromise({
+      try: () => lucid.utxosAt(params.settlementAddress),
+      catch: (err) =>
+        new LucidError({
+          message: "Failed to fetch Settlement UTxOs",
+          cause: err,
+        }),
+    });
+    if (settlementAllUtxos.length === 0) {
+      yield* new LucidError({
+        message: "Failed to build the Settlement transaction",
+        cause: "No UTxOs found in Settlement contract address",
+      });
+    }
+    const settlementInputUtxo =
+      yield* getSettlementUTxOWithoutClaim(settlementAllUtxos);
+
+    const spendDatum: SettlementDatum = {
+      depositsRoot: settlementInputUtxo.datum.depositsRoot,
+      withdrawalsRoot: settlementInputUtxo.datum.withdrawalsRoot,
+      transactionsRoot: settlementInputUtxo.datum.transactionsRoot,
+      resolutionClaim: {
+        resolutionTime: params.newBondUnlockTime,
+        operator: params.resolutionClaimOperator,
+      },
+    };
+    const spendDatumCBOR = Data.to(spendDatum, SettlementDatum);
```

Comment:
Better naming to help readability. Also, if you can use the spread operator, it'll be more concise:
```suggestion
    const updatedDatum: SettlementDatum = {
      ...settlementInputUtxo.datum,
      resolutionClaim: {
        resolutionTime: params.newBondUnlockTime,
        operator: params.resolutionClaimOperator,
      },
    };
    const updatedDatumCBOR = Data.to(spendDatum, SettlementDatum);
```